r/tests/testthat/test-dataset-csv.R

```diff
@@ -288,3 +289,12 @@ test_that("Column names inferred from schema for headerless CSVs (ARROW-14063)",
   ds <- open_dataset(headerless_csv_dir, format = "csv", schema = schema(int = int32(), dbl = float64()))
   expect_equal(ds %>% collect(), tbl)
 })
+
+test_that("open_dataset() deals with BOMs (byte-order-marks) correctly", {
+  writeLines("\xef\xbb\xbfa,b\n1,2\n", con = csv_file)
+
+  expect_equal(
+    open_dataset(csv_file, format = "csv") %>% collect(),
+    tibble(a = 1, b = 2)
+  )
```

Comment:
Would it be possible to make two csv files and read those in as a dataset? This _should_ work just the same, but it would be good to confirm that the second csv has the proper BOM handling as well.

Reply:
Oh that's a good idea. I'll add that really quick.

Reply:
@wjones127 @jonkeane I think the additional tests are now failing. I got [this](https://github.com/apache/arrow/runs/4470780115?check_suite_focus=true) in a different PR. I haven't yet rebased so I don't even have those tests in my branch.

Reply:
@dragosmg I filed https://issues.apache.org/jira/browse/ARROW-15041